internal/tui/portforward_handlers.go

```diff
@@ -1,175 +1,234 @@
+//     its initial status, a stop channel (if successful), and any error encountered during setup.
+//
+// Returns the updated model and a nil command as no further async operations are directly initiated here.
+func handlePortForwardSetupCompletedMsg(m model, msg portForwardSetupCompletedMsg) (model, tea.Cmd) {
 	if pf, ok := m.portForwards[msg.label]; ok {
-		errMsgText := fmt.Sprintf("[%s %s ERROR] %s", msg.label, msg.streamType, msg.err.Error())
-		pf.err = msg.err
-		pf.output = append(pf.output, "ERROR: "+msg.err.Error())
-		m.combinedOutput = append(m.combinedOutput, errMsgText)
-
-		// Update status unless it's a more specific startup/restart failure message
-		if pf.statusMsg != "Failed to start" && pf.statusMsg != "Restart failed" {
-			pf.statusMsg = "Error"
-		}
-
-		// Mark streams as closed based on the error message context
-		// This is a heuristic; specific errors might imply specific streams closed.
-		// For a general error, or if streamType is "general", it might not close them.
-		if msg.streamType == "stdout" {
-			pf.stdoutClosed = true
-		} else if msg.streamType == "stderr" {
-			pf.stderrClosed = true
-		} else if msg.streamType == "general" {
-			// If it's a general error (e.g. process failed to start), both streams are effectively closed.
-			pf.stdoutClosed = true
-			pf.stderrClosed = true
-			pf.active = false // Mark as inactive if the process couldn't start or failed critically
+		if msg.err != nil { // Error during synchronous setup in StartPortForwardClientGo
+			pf.err = msg.err
+			// msg.status might be empty if error was very early, or could be a partial status.
+			// It's safer to construct a clear error status.
+			pf.statusMsg = fmt.Sprintf("Setup Failed: %v", msg.err)
+			pf.active = false
+			pf.stopChan = nil
+			m.combinedOutput = append(m.combinedOutput, fmt.Sprintf("[%s ERROR] Port-forward direct setup failed: %v. Async process not started.", msg.label, msg.err))
+		} else {
+			// Synchronous setup in StartPortForwardClientGo was successful.
+			// msg.status contains the initial status log (e.g., "Initializing...").
+			pf.stopChan = msg.stopChan
+			pf.statusMsg = msg.status // Set initial status for TUI display
+			pf.err = nil
+			pf.active = true
+			// The sendUpdate call within StartPortForwardClientGo also sent this initialStatus for logging.
+			m.combinedOutput = append(m.combinedOutput, fmt.Sprintf("[%s] Port-forward async setup initiated. Initial TUI status: %s", msg.label, msg.status))
 		}
 
-		// Trim outputs
+		// Trim combined output
 		if len(m.combinedOutput) > maxCombinedOutputLines {
 			m.combinedOutput = m.combinedOutput[len(m.combinedOutput)-maxCombinedOutputLines:]
 		}
-		if len(pf.output) > maxCombinedOutputLines {
-			pf.output = pf.output[len(pf.output)-maxCombinedOutputLines:]
-		}
+	} else {
+		m.combinedOutput = append(m.combinedOutput, fmt.Sprintf("[TUI WARNING] No PF found for Lbl['%s'] during SetupCompleted.", msg.label))
```

Comment:
Avoid cryptic message

```suggestion
		m.combinedOutput = append(m.combinedOutput, fmt.Sprintf("[TUI WARNING] No Port-forward found for label['%s'] during SetupCompleted.", msg.label))
```